lib/galaxy/managers/collections.py

```diff
@@ -52,11 +54,20 @@
 )
 from galaxy.util import validation
 
+if TYPE_CHECKING:
+    from galaxy.model import (
+        DatasetCollectionInstance,
+        HistoryDatasetAssociation,
+    )
+    from galaxy.tool_util_models.tool_source import FieldDict
+
 log = logging.getLogger(__name__)
 
 ERROR_INVALID_ELEMENTS_SPECIFICATION = "Create called with invalid parameters, must specify element identifiers."
 ERROR_NO_COLLECTION_TYPE = "Create called without specifying a collection type."
 
+HDCAElementType = Union[DatasetCollection, "HistoryDatasetAssociation"]
```

Comment:
```suggestion
HDCAElementObjectType = Union[DatasetCollection, "HistoryDatasetAssociation"]
```
And technically it could also be LibraryDatasetDatasetAssociation ... though I'm sure this would break in many places.

Reply:
Changed the type name in a rebase.
Agreed it could be `HDCAElementObjectType = Union[DatasetCollection, "DatasetInstance"]` in the future, but I think it's better to reflect reality at the moment.